File: isegoria/src/main/java/com/eulersbridge/isegoria/personality/PersonalityPermissionFragment.java
package com.eulersbridge.isegoria.personality;

import android.arch.lifecycle.ViewModelProviders;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import com.eulersbridge.isegoria.R;

public class PersonalityPermissionFragment extends Fragment {

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.personality_screen1_fragment, container, false);

        //noinspection ConstantConditions
        PersonalityViewModel viewModel = ViewModelProviders.of(getActivity()).get(PersonalityViewModel.class);

        final Button takePersonalityButton = rootView.findViewById(R.id.takePersonalityButton);
        takePersonalityButton.setOnClickListener(view -> viewModel.setUserContinuedQuestions());

        Button skipPersonalityQuestions = rootView.findViewById(R.id.skipPersonality);
        skipPersonalityQuestions.setOnClickListener(v -> viewModel.setUserSkippedQuestions());

        return rootView;
    }
}

File: isegoria/src/main/java/com/eulersbridge/isegoria/personality/PersonalityPermissionFragment.kt
package com.eulersbridge.isegoria.personality

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.eulersbridge.isegoria.R
import kotlinx.android.synthetic.main.personality_screen1_fragment.*

class PersonalityPermissionFragment : Fragment() {

    private lateinit var viewModel: PersonalityViewModel

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View
        = inflater.inflate(R.layout.personality_screen1_fragment, container, false)

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        takePersonalityButton.setOnClickListener { this.onContinue() }
    }

    fun setViewModel(viewModel: PersonalityViewModel) {
        this.viewModel = viewModel
    }

    private fun onContinue() {
        viewModel.questionsContinued.value = true
    }
}
